boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -650,6 +657,59 @@ LoadWindowsCore(IN USHORT OperatingSystemVersion,
     return Success;
 }
 
+static
+BOOLEAN
+WinLdrInitErrataInf(IN PCSTR SystemRoot)
+{
+    CHAR ErrataFilePath[1024];
+    ARC_STATUS Status;
+    ULONG FileId;
+    FILEINFORMATION FileInfo;
+    PVOID PhysicalBase;
+    ULONG FileSize, BytesRead;
+
+    RtlStringCbCopyA(ErrataFilePath, sizeof(ErrataFilePath), SystemRoot);
+    RtlStringCbCatA(ErrataFilePath, sizeof(ErrataFilePath), "inf\\errata.inf");
+
+    Status = ArcOpen((PCHAR)ErrataFilePath, OpenReadOnly, &FileId);
+    if (Status != ESUCCESS)
+    {
+        WARN("Error while opening errata.inf: %u\n", Status);
+        return FALSE;
+    }
+
+    Status = ArcGetFileInformation(FileId, &FileInfo);
+    if (Status != ESUCCESS)
+    {
+        WARN("Error while getting file information for errata.inf: %u\n", Status);
+        ArcClose(FileId);
+        return FALSE;
+    }
+    FileSize = FileInfo.EndingAddress.LowPart;
+
+    PhysicalBase = MmAllocateMemoryWithType(FileSize, LoaderRegistryData);
```

Comment:
The pointer is passed in the `LOADER_PARAMETER_EXTENSION` of the `LOADER_PARAMETER_BLOCK`, so usually you need to specify an explicit memory type.
Also, on Windows Server 2003, you get this:
![image](https://user-images.githubusercontent.com/1969829/66262871-40149e00-e7e9-11e9-8c39-b7e3b7b27103.png)
(this is my hand-made analysis because I didn't have complete required symbols to do that straightforwardly in WinDbg, following the example provided in https://books.google.hr/books?id=3rdCAwAAQBAJ&pg=PA515&lpg=PA515).
The memory type "19" is `LoaderRegistryData`, according to https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/arc/arc.h;hb=9ad0dd18562293631d40e0b3d32646a8e14ad4ca#l125 .